### ISSUE TITLE
marshall_for_signature for entry_commit and chain_commit

### ISSUE THREAD

**Issue by samuelvanderwaal** [sambarnes/factom-core#17]
It looks like both EntryCommit and ChainCommit classes could use a `marshall_for_signature` method to aid usability per your comments on Discord. 

To do this it looks like I should:

* Give `signature` a default value of None
* Write a `marshall_for_signature` method similar to the one in the `FactoidTransaction` class that creates the bytes to be signed
* Implement `signature` as a property with a setter so it can be set, or create a `set_signature` method

This would lead to a user story something like this:

```Python
chain_commit = ChainCommit(timestamp, chain_id_hash, commit_weld, entry_hash, ec_spent, ec_public_key)
signature = signer.sign(chain_commit.marshall_for_signature())
chain_commit.signature = signature
```

Let me know if this is in-line what you're thinking or if I should go a different path.

**Comment by sambarnes** [comment]
Yep, I think that sounds exactly right! This would be a very welcome addition and will allow users of the factomd RPC python client library to optionally drop the factom-walletd dependency for creating chains/entries.

Let's keep the changes limited to only that and don't worry about adding extra error handling or things like that since I still need to decide on the conventions for this library to follow. Trying to keep it as light-weight as possible for now and only add things as they are truly needed to make the structures usable.

**Comment by samuelvanderwaal** [comment]
Great, I'll knock this out tomorrow.

**Comment by sambarnes** [comment]
Resolved by #18
